source/common/protobuf/utility.cc

```diff
@@ -439,6 +464,19 @@ class UnexpectedFieldProtoVisitor : public ProtobufMessage::ConstProtoVisitor {
 
     // If this field is deprecated, warn or throw an error.
     if (field.options().deprecated()) {
+      if (absl::StartsWith(field.name(), Config::VersionUtil::DeprecatedFieldShadowPrefix)) {
+        // The field was marked as hidden_envoy_deprecated and an error must be thrown
+        // unless it is part of an explicit test that needs access to the deprecated field
+        if (!runtime_ ||
+            !runtime_->snapshot().deprecatedFeatureEnabled(
+                absl::StrCat("envoy.deprecated_features:", field.full_name()), false)) {
```

Comment:
Let me know if this can be clearer (I thought the comment made sense,  but it might need to be improved)

Reply:
It's probably worth leaving a note such as "When we enable universal runtime deprecation override, this must not be applied here, since we only want point field overrides for test.".